### DISCUSSION TITLE
SolaX HTTP API integration

### DISCUSSION THREAD

**Discussion by PatrikTrestik** [wills106/homeassistant-solax-modbus#636] [Ideas]
I'm starting this discussion to clarify whether it is feasible to integrate HTTP API as source of data apart from Modbus.

I created POC for EV Changer here https://github.com/PatrikTrestik/homeassistant-solax-modbus/tree/feature/solax-ev-http-api 
It is very dirty stitched together and requires Modbus register to HTTP API mapping.

If there is will to keep HTTP API here, my first suggestion is to create separate plugins for API variants.

**Comment by PatrikTrestik** [comment]
@wills106 I would like to hear your opinion on this. 
I'll have some time for private projects in next week and I can start something.

What I have, what works and what not:
- Working HTTP API communication for EV Charger which can be also used with new WiFi Dongle and Inverter.
- Mapping for most of EV Charger functions. Unfortunately nothing corresponds to Modbus and has to me found by experiments.
- Dirty integration into this project. It is using fake Modbus mapping and is not  feasible to stay as it is. New plugin for API has to be created and this moved there.

**Reply by cleveHEX** [reply]
Maybe if nothing corresponds to modbus, a separate integration would be in place? Just my opinion. This would give you more control and quick development.

**Reply by PatrikTrestik** [reply]
Actually yes, and no. Here is good community around solax operation and new integration will start lots of discussions over again.
Also there is good part of work which can be reused. 

I'll try to stay here and create new plugin which does not use modbus part. I'll see.

**Reply by wills106** [reply]
I agree it might be easier to have it separate, but it would also be nice to include EV Charger Control when in Smart Functions mode.

Although Ideally I would like to emulate the Smart Functions so Inverters like my Gen3 Hybrid can communicate with the Charger.

**Reply by PatrikTrestik** [reply]
I'm working on separate integration. I keep it close to this one, keeping most of code similar. I hope that it will help to apply changes regarding hardware to be propagated from here to my integration easily.

I'll try to direct any hardware related questions to here to keep it on one place.

**Reply by PatrikTrestik** [reply]
All HTTP related stuff moved here: https://github.com/PatrikTrestik/homeassistant-solax-http

**Comment by 00feichti** [comment]
Hallo,
i am not the best Homeassitant User.
can you write how to ad your mod to Homeassistant?

i woud like to add my EV Charger to my Homeassistant

Thank you

**Reply by PatrikTrestik** [reply]
Hi @00feichti , I can write description here. But because you admitted that you are not experienced user, I would not suggest it.
This project is in prove of concept at the moment, can be full of bugs and can get your home assistant inoperable.

You have to know how to copy files to your HA instance under config/custom_components.
Also create full backup and find information how to fully restore your HA.

Sorry, but I cannot take any responsibility for correct state of code. It needs more work to be User ready.
But I'm running it on my instance, fixing all bugs I found.

Also there is significant architecture change required. If you use it, it might be required to completely reconfigure in future.

You have been warned. 
Files are here: https://github.com/PatrikTrestik/homeassistant-solax-modbus/tree/feature/solax-ev-http-api
Take content of custom_components/solax_modbus and copy it over to your HA.

**Reply by PatrikTrestik** [reply]
Please use this: https://github.com/PatrikTrestik/homeassistant-solax-http
Lots of stuff simplified and bugs fixed.
